core/src/client/mod.rs

```diff
@@ -675,4 +703,540 @@ impl StarknetClient for StarknetClientImpl {
+                            &"0xac91334ba861cb94cba2b1fd63df7e87c15ca73666201abd10b5462255a5c642"
+                                .as_bytes()[1..33],
+                        );
+                        let receipts_root = PrimitiveH256::from_slice(
+                            &"0xf2c8755adf35e78ffa84999e48aba628e775bb7be3c70209738d736b67a9b549"
+                                .as_bytes()[1..33],
+                        );
+
+                        let number = U256::from(block_with_txs.block_number);
+                        let timestamp = U256::from(block_with_txs.timestamp);
+
+                        let blockhash_opt = Some(PrimitiveH256::from_slice(
+                            &(block_with_txs.block_hash).to_bytes_be(),
+                        ));
+                        let blocknum_opt = Some(U256::from(block_with_txs.block_number));
+
+                        let transactions = self
+                            .filter_transactions(
+                                block_with_txs.transactions,
+                                blockhash_opt,
+                                blocknum_opt,
+                            )
+                            .await?;
+
+                        let header = Header {
+                            hash: Some(hash),
+                            parent_hash,
+                            uncles_hash: parent_hash,
+                            author: sequencer,
+                            miner: sequencer,
+                            state_root,
+                            // BlockWithTxHashes doesn't have a transactions root
+                            transactions_root,
+                            // BlockWithTxHashes doesn't have a receipts root
+                            receipts_root,
+                            number: Some(number),
+                            gas_used,
+                            gas_limit,
+                            extra_data,
+                            logs_bloom,
+                            timestamp,
+                            difficulty,
+                            nonce,
+                            size,
+                            base_fee_per_gas,
+                            mix_hash,
+                        };
+                        let block = Block {
+                            header,
+                            total_difficulty,
+                            uncles: vec![],
+                            transactions,
+                            base_fee_per_gas: None,
+                            size,
+                        };
+                        Ok(Rich::<Block> {
+                            inner: block,
+                            extra_info: BTreeMap::default(),
+                        })
+                    }
+                }
+            }
+        }
+    }
+    async fn filter_transactions(
+        &self,
+        initial_transactions: Vec<StarknetTransaction>,
+        blockhash_opt: Option<PrimitiveH256>,
```

Comment:
I think this does not have to be an option, since you have transactions as input, they must come from a specific block, meaning you can pass the blockhash and block_number as "sure" params, not options

Reply:
makes senses

Reply:
alright